### PR TITLE
[PLATFORM-1332] Static Assets Service - cassandra integration

### PR DESCRIPTION
**Pull request by jcellary**
Added storage and retrieving of metadata from cassandra when storing image assets.

Also changed the image url to not include extension and added a gradle task to recreate cassandra keyspace. 

Pinging @sqreek

### CODE REVIEW

File: service/static-assets/src/main/java/com/wikia/services/staticassets/StaticAssetsConfiguration.java [hunk 1]
```diff
@@ -53,4 +53,8 @@
   @JsonProperty
   @NotNull
   private Integer cephPort;
```
Comment (WikiaTech) [inline service/static-assets/src/main/java/com/wikia/services/staticassets/StaticAssetsConfiguration.java]: MAJOR Remove this unused "cephPort" private field. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1068)

File: service/static-assets/src/main/java/com/wikia/services/staticassets/StaticAssetsConfiguration.java [hunk 2]
```diff
@@ -53,4 +53,8 @@
...
+
+  @JsonProperty
+  @NotNull
+  private String imageS3BucketName;
```
Comment (WikiaTech) [inline service/static-assets/src/main/java/com/wikia/services/staticassets/StaticAssetsConfiguration.java]: MAJOR Remove this unused "imageS3BucketName" private field. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1068)

File: service/static-assets/src/main/java/com/wikia/services/staticassets/resources/StaticAssetsResource.java [hunk 3]
```diff
@@ -46,9 +52,23 @@
 
   private final ImageService imageService;
 
+  private final Session session;
```
Comment (WikiaTech) [inline service/static-assets/src/main/java/com/wikia/services/staticassets/resources/StaticAssetsResource.java]: MAJOR Remove this unused "session" private field. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1068)

File: service/static-assets/src/main/java/com/wikia/services/staticassets/models/cassandra/Asset.java [hunk 4]
```diff
@@ -31,45 +31,28 @@
   private String originalFilename;
   @NotNull
   @Column
+  private String fileFormat;
```
Comment (WikiaTech) [inline service/static-assets/src/main/java/com/wikia/services/staticassets/models/cassandra/Asset.java]: MAJOR Remove this unused "fileFormat" private field. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1068)

File: service/static-assets/src/main/java/com/wikia/services/staticassets/models/cassandra/Asset.java [hunk 5]
```diff
@@ -31,45 +31,28 @@
...
+  @NotNull
+  @Column
   private String ownerId;
```
Comment (WikiaTech) [inline service/static-assets/src/main/java/com/wikia/services/staticassets/models/cassandra/Asset.java]: MAJOR Remove this unused "ownerId" private field. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1068)

File: service/static-assets/src/main/java/com/wikia/services/staticassets/models/cassandra/Asset.java [hunk 6]
```diff
@@ -13,10 +13,10 @@
...
 @ToString
-@Builder
 @EqualsAndHashCode
+@Data
 @Table(name = "assets")
-public class StoredAsset {
+public class Asset {
   @NotNull
   @PartitionKey
   private UUID uuid;
```
Comment (WikiaTech) [inline service/static-assets/src/main/java/com/wikia/services/staticassets/models/cassandra/Asset.java]: MAJOR Remove this unused "uuid" private field. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1068)

File: service/static-assets/src/main/java/com/wikia/services/staticassets/models/cassandra/Asset.java [hunk 7]
```diff
@@ -31,45 +31,28 @@
   private String originalFilename;
```
Comment (WikiaTech) [inline service/static-assets/src/main/java/com/wikia/services/staticassets/models/cassandra/Asset.java]: MAJOR Remove this unused "originalFilename" private field. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1068)

File: service/static-assets/src/main/java/com/wikia/services/staticassets/commands/SetupDb.java [hunk 8]
```diff
@@ -24,6 +24,10 @@ public SetupDb(Application<StaticAssetsConfiguration> application) {
   @Override
   protected void run(Environment environment, Namespace namespace,
                      StaticAssetsConfiguration configuration) throws Exception {
+    String recreateStr = System.getProperty("cassandra.recreate");
+    if (recreateStr != null && Boolean.parseBoolean(recreateStr)) {
+      keyspaceHelper.dropKeyspace();
+    }
     keyspaceHelper.ensureKeyspaceExists();
     session.execute(Cassandra.CREATE_ASSET_TABLE);
     System.exit(0);
```
Comment (WikiaTech) [inline service/static-assets/src/main/java/com/wikia/services/staticassets/commands/SetupDb.java]: CRITICAL Remove this call to "exit" or ensure it is really required. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1147)

File: service/static-assets/src/test/java/com/wikia/services/staticassets/resources/StaticAssetsResourceTest.java [hunk 9]
```diff
@@ -42,18 +49,57 @@
...
+
+  public Mapper<Asset> assetMapperMock;
+
+  private final String testBucketName = "test-bucket";
+
+  private final String testUuid = "12345678-1234-1234-1234-123456789012";
+
+  private final String testImageUrl = String.format("/images/%s", testUuid);
+
+  private final byte[] testFileContent = new byte[]{1, 2, 3, 4};
```
Comment (WikiaTech) [inline service/static-assets/src/test/java/com/wikia/services/staticassets/resources/StaticAssetsResourceTest.java]: MAJOR Make this final field static too. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1170)

File: service/static-assets/src/test/java/com/wikia/services/staticassets/resources/StaticAssetsResourceTest.java [hunk 10]
```diff
@@ -42,18 +49,57 @@
 
   public ImageService imageServiceMock;
```
Comment (WikiaTech) [inline service/static-assets/src/test/java/com/wikia/services/staticassets/resources/StaticAssetsResourceTest.java]: MAJOR Make imageServiceMock a static final constant or non-public and provide accessors if needed. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)

File: service/static-assets/src/test/java/com/wikia/services/staticassets/resources/StaticAssetsResourceTest.java [hunk 11]
```diff
@@ -42,18 +49,57 @@
...
 
+  public Session sessionMock;
```
Comment (WikiaTech) [inline service/static-assets/src/test/java/com/wikia/services/staticassets/resources/StaticAssetsResourceTest.java]: MAJOR Make sessionMock a static final constant or non-public and provide accessors if needed. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)

File: service/static-assets/src/test/java/com/wikia/services/staticassets/resources/StaticAssetsResourceTest.java [hunk 12]
```diff
@@ -42,18 +49,57 @@
...
+
+  public MappingManager mappingManagerMock;
```
Comment (WikiaTech) [inline service/static-assets/src/test/java/com/wikia/services/staticassets/resources/StaticAssetsResourceTest.java]: MAJOR Make mappingManagerMock a static final constant or non-public and provide accessors if needed. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)

File: service/static-assets/src/test/java/com/wikia/services/staticassets/resources/StaticAssetsResourceTest.java [hunk 13]
```diff
@@ -42,18 +49,57 @@
...
+
+  public Mapper<Asset> assetMapperMock;
```
Comment (WikiaTech) [inline service/static-assets/src/test/java/com/wikia/services/staticassets/resources/StaticAssetsResourceTest.java]: MAJOR Make assetMapperMock a static final constant or non-public and provide accessors if needed. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)

File: service/static-assets/src/test/java/com/wikia/services/staticassets/resources/StaticAssetsResourceTest.java [hunk 14]
```diff
@@ -42,18 +49,57 @@
...
+
+  private final String testBucketName = "test-bucket";
```
Comment (WikiaTech) [inline service/static-assets/src/test/java/com/wikia/services/staticassets/resources/StaticAssetsResourceTest.java]: MAJOR Make this final field static too. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1170)

File: service/static-assets/src/test/java/com/wikia/services/staticassets/resources/StaticAssetsResourceTest.java [hunk 15]
```diff
@@ -42,18 +49,57 @@
...
+
+  private final String testUuid = "12345678-1234-1234-1234-123456789012";
```
Comment (WikiaTech) [inline service/static-assets/src/test/java/com/wikia/services/staticassets/resources/StaticAssetsResourceTest.java]: MAJOR Make this final field static too. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1170)

File: service/static-assets/src/main/java/com/wikia/services/staticassets/utils/Cassandra.java [hunk 16]
```diff
@@ -5,16 +5,13 @@
 
 public class Cassandra {
```
Comment (WikiaTech) [inline service/static-assets/src/main/java/com/wikia/services/staticassets/utils/Cassandra.java]: MAJOR Add a private constructor to hide the implicit public one. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1118)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 26 issues:
* CRITICAL 1 critical
* MAJOR 25 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Remove this unused "cephAccessKey" private field.](https://github.com/Wikia/pandora/blob/9ea15415a2a8447e6080cdc5a34c5b448c65f916/service/static-assets/src/main/java/com/wikia/services/staticassets/StaticAssetsConfiguration.java#L39) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1068)
* MAJOR [Remove this unused "defaultCassandraKeyspace" private field.](https://github.com/Wikia/pandora/blob/9ea15415a2a8447e6080cdc5a34c5b448c65f916/service/static-assets/src/main/java/com/wikia/services/staticassets/StaticAssetsConfiguration.java#L35) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1068)
* MAJOR [Remove this unused "cassandraConfiguration" private field.](https://github.com/Wikia/pandora/blob/9ea15415a2a8447e6080cdc5a34c5b448c65f916/service/static-assets/src/main/java/com/wikia/services/staticassets/StaticAssetsConfiguration.java#L32) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1068)
* MAJOR [Remove this unused "cephSecretKey" private field.](https://github.com/Wikia/pandora/blob/9ea15415a2a8447e6080cdc5a34c5b448c65f916/service/static-assets/src/main/java/com/wikia/services/staticassets/StaticAssetsConfiguration.java#L43) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1068)
* MAJOR [Remove this unused "cephDomain" private field.](https://github.com/Wikia/pandora/blob/9ea15415a2a8447e6080cdc5a34c5b448c65f916/service/static-assets/src/main/java/com/wikia/services/staticassets/StaticAssetsConfiguration.java#L51) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1068)
* MAJOR [Remove this unused "httpClientConfiguration" private field.](https://github.com/Wikia/pandora/blob/9ea15415a2a8447e6080cdc5a34c5b448c65f916/service/static-assets/src/main/java/com/wikia/services/staticassets/StaticAssetsConfiguration.java#L26) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1068)
* MAJOR [Remove this unused "cephEndpoint" private field.](https://github.com/Wikia/pandora/blob/9ea15415a2a8447e6080cdc5a34c5b448c65f916/service/static-assets/src/main/java/com/wikia/services/staticassets/StaticAssetsConfiguration.java#L47) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1068)
* MAJOR [Remove this unused "swaggerConfiguration" private field.](https://github.com/Wikia/pandora/blob/9ea15415a2a8447e6080cdc5a34c5b448c65f916/service/static-assets/src/main/java/com/wikia/services/staticassets/StaticAssetsConfiguration.java#L29) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1068)
* MAJOR [Remove this unused "cephPath" private field.](https://github.com/Wikia/pandora/blob/9ea15415a2a8447e6080cdc5a34c5b448c65f916/service/static-assets/src/main/java/com/wikia/services/staticassets/models/cassandra/Asset.java#L28) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1068)
* MAJOR [Remove this unused "cephBucket" private field.](https://github.com/Wikia/pandora/blob/9ea15415a2a8447e6080cdc5a34c5b448c65f916/service/static-assets/src/main/java/com/wikia/services/staticassets/models/cassandra/Asset.java#L25) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1068)

**Comment by sqreek** [comment]
LGTM, we need to remove rule "Remove this unused" from sonar for Lombok classes

**Pull request merged**